scheduler/src/cook/tools.clj

```diff
@@ -916,27 +916,50 @@
        first
        field))
 
-(defn filter-based-on-quota
+(defn get-quota-for-pool
+  "Given a pool name, determine the quota for that pool."
+  [quotas effective-pool-name]
+  (match-based-on-pool-name quotas effective-pool-name :quota))
+
+(defn filter-based-on-user-quota
   "Lazily filters jobs for which the sum of running jobs and jobs earlier in the queue exceeds one of the constraints,
    max-jobs, max-cpus or max-mem"
   [user->quota user->usage queue]
   (letfn [(filter-with-quota [user->usage job]
             (let [user (:job/user job)
                   job-usage (job->usage job)
                   user->usage' (update-in user->usage [user] #(merge-with + job-usage %))]
-              (log/debug "Quota check" {:user user
-                                        :usage (get user->usage' user)
-                                        :quota (user->quota user)})
+              (log/debug "User quota check" {:user user
+                                             :usage (get user->usage' user)
+                                             :quota (user->quota user)})
               [user->usage' (below-quota? (user->quota user) (get user->usage' user))]))]
     (filter-sequential filter-with-quota user->usage queue)))
 
+(defn filter-based-on-pool-quota
+  "Lazily filters jobs for which the sum of running jobs and jobs earlier in the queue exceeds one of the constraints,
+   max-jobs, max-cpus or max-mem"
+  [quota usage queue]
+  (log/debug "Pool quota and usage:" {:quota quota :usage usage})
+  (if (nil? quota)
+    queue
+    (letfn [(filter-with-quota [usage job]
+              (let [job-usage (job->usage job)
+                    usage' (merge-with + job-usage usage)]
+                (log/debug "Pool quota check" {:usage usage'
+                                               :quota quota})
+                [usage' (below-quota? quota usage')]))]
+      (filter-sequential filter-with-quota usage queue))))
```

Comment:
It's hard to tell what the shapes of `quota` and `usage` are. Is it `pool -> {:cpus cpus :mem mem}`? Can we make the names more descriptive (and maybe describe the shapes in the docstring)?

Reply:
Docstring updated. Not sure what would be more descriptive names. I'm happy to rename if you have a proposal.

Reply:
Docstring is helpful, thanks